inspirehep/modules/workflows/utils/__init__.py

```diff
@@ -158,6 +158,40 @@ def _try_to_log(logfn, *args, **kwargs):
     return _decorator
 
 
+def do_not_repeat(func, step_id):
+    """Skip the given workflow's step after the first run.
+
+    Sets in `obj.extra_data` a dictionary named `skipped_steps`, where the key
+    is the name of the workflow's step to skip after the first run, and the
+    value is False if the step has never been skipped, otherwise True.
+
+    Args:
+        func (object): the workflow's step to skip from the second run on
+        step_id (string): an identifier for the step to be skipped
+
+    Return:
+        a decorator function
+    """
+    @with_debug_logging
+    @wraps(func)
+    def _dont_repeat(*args, **kwargs):
+        obj = args[0]
+        if step_id in obj.extra_data.setdefault('skipped_steps', {}).keys():
+            obj.log.info(
+                'Step "%s" already executed in a previous workflow run' %
+                step_id
+            )
+            obj.extra_data['skipped_steps'][step_id] = True
+            obj.save()
+            return
+        else:
+            func(*args, **kwargs)
+            obj.extra_data['skipped_steps'][step_id] = False
+            obj.save()
```

Comment:
nono, this is not good, you should use a "parametrized decorator", like:
```python

def my_parameterized_decorator(param1):
    def _real_decorator(func):
        @wraps(func)
        def _wrapper_func(*args, **kwargs):
            print('param1 was %s' % param1)
            if _already_run(param1):
                return

            return func(*args, **kwargs)

    return _real_decorator

## then you can use it like
@prametrized_decorator(param1="something')
def myfunc(some, args, and_kwargs=None):
    pass
```

Reply:
That way you don't have to mangle the args or anything, and it's totally agnostic of the function inside.
You should also return whatever the function returns (just in case, very recommended practice).

Reply:
the api would be nicer if you could use it both ways, but we don't want that for this one, and it makes the code a bit uglier (a big if to check `callable` for example).